src/iso15118/d20/state/dc_charge_loop.cpp

```diff
@@ -18,6 +18,24 @@ using Dynamic_BPT_DC_Req = message_20::DC_ChargeLoopRequest::BPT_Dynamic_DC_CLRe
 using Scheduled_DC_Res = message_20::DC_ChargeLoopResponse::Scheduled_DC_CLResControlMode;
 using Scheduled_BPT_DC_Res = message_20::DC_ChargeLoopResponse::BPT_Scheduled_DC_CLResControlMode;
 
+static auto fill_parameters(const message_20::DisplayParameters& req_parameters) {
+    auto parameters = session::feedback::DisplayParameters{};
+
+    parameters.present_soc = req_parameters.present_soc;
+    parameters.minimum_soc = req_parameters.min_soc;
+    parameters.target_soc = req_parameters.target_soc;
+    parameters.maximum_soc = req_parameters.max_soc;
+    parameters.remaining_time_to_minimum_soc = req_parameters.remaining_time_to_min_soc;
+    parameters.remaining_time_to_target_soc = req_parameters.remaining_time_to_target_soc;
+    parameters.remaining_time_to_maximum_soc = req_parameters.remaining_time_to_max_soc;
+    if (req_parameters.battery_energy_capacity.has_value()) {
```

Comment:
depending on style and convention `if (req_parameters.battery_energy_capacity)` is also possible.  It is shorter but not that explicit.